[CS2113T-T09-4] Modganiser

geraldkoh4 opened the pull request:
Modganiser is effortless module planning, at your fingertips via the Command Line Interface (CLI). It helps Information security students graduate in time by giving you a platform to plan all 4 years of your modules to ensure that you meet graduation requirements (UE/GEs).

Koh-Jing-Jie-Marcus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+will call `findModuleByName`. This method will then print out all the modules that have the keyword "CS" in their
+module name.
+
+The following sequence diagram shows how the `findModuleByName` operation works:
```
Alt frame exceeding findModuleByName frame

Koh-Jing-Jie-Marcus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+
+The following sequence diagram shows how the `findModuleByName` operation works:
+
+![FindModule](diagrams/FindModule.png)
```
No bars used for self invocation

Koh-Jing-Jie-Marcus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+7) If the user wants to update the Semester, the `editSemester()` method is called.
+   This method directly modifies the `semester` attribute of the `Module` object.
+
+![editCommand](diagrams/EditCommand.png)
```
No bars used for self invocation

Koh-Jing-Jie-Marcus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+
+`Manpage#printManPageExit()` - Prints description of how to use the exit feature.
+
+![ManPage](diagrams/ManPage.png)
```
Can improve readability, words overlapping with bars

Koh-Jing-Jie-Marcus commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+[Introduction](#introduction)
+
+[Table of contents](#table-of-contents) 
+0. [Acknowledgements](#acknowledgements)
```
formatting issue

pyongjie commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+
+The sequence of events above can be represented with the following sequence diagram:
+
+![addInputCommand](diagrams/AddInputCommand.png)
```
Perhaps could have more minimal notations in your sequence diagram? Personally, I feel the descriptions for the return arrows from the calling of the constructors of various classes could be omitted as it does not result in loss of information. This could make the diagram more concise and easier to read.

Sherlock-YH replied:
I like this diagram

SongZijin replied:
Since you are calling a constructor to create the Core / GeneralElective / etc., they should be created with a construction block, instead of just being called with activation block.
<img width="64" alt="Screenshot 2023-03-30 at 3 39 03 PM" src="https://user-images.githubusercontent.com/88227764/228764345-bfaf7acd-9f77-431f-8fa6-4d5e9ac11f12.png">

thant commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+**Design Considerations**
+
+**Alternative 1** : Prints the manual page individually as features. 
+Pros: Avoids wordy and long manual page when command is man, ability to have more description
```
![image](https://user-images.githubusercontent.com/1620654/228721882-add2d1bc-188d-4efe-a199-104ddf7d7cc7.png)
Pros and cons not appearing as new lines in the DG.

thant commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+`ManPage#printManPageExit()` - Prints description of how to use the exit feature.
+
+![ManPage](diagrams/ManPage.png)
+
+**Design Considerations**
+
+**Alternative 1** : Prints the manual page individually as features. 
+Pros: Avoids wordy and long manual page when command is man, ability to have more description
+for specific features. Avoids long methods. 
+Cons: Extra step of typing another command to understand a feature well.
+
+**Alternative 2**: Combining all feature descriptions in a single manual page. 
+Pros: Avoid extra step of typing another command to understand a feature well.
+Cons: Wordy and long manual page when command is man, user has to scroll through
```
![image](https://user-images.githubusercontent.com/1620654/228721988-93e04f1e-af7f-4636-8aa2-b0f3733c321f.png)

Similarly the pros and cons aren't on new lines, and perhaps the command is man should highlight that "man" as a code input

BernardLesley replied:
Great job adding pros an cons in your design considerations, however, I think the pros and cons can start on a new line to improve readability

thant reviewed (commented):
Overall, shows great promise once missing sections are filled in with the relevant information.

sriram-senthilkr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+
+The following sequence diagram shows how the `findModuleByName` operation works:
+
+![FindModule](diagrams/FindModule.png)
```
Bar ends before the return of self invocation method
<img width="512" alt="image" src="https://user-images.githubusercontent.com/88694646/228722901-675d1502-31d6-4f3a-9a98-50bc8bddb016.png">

BernardLesley replied:
<img width="183" alt="image" src="https://user-images.githubusercontent.com/67412977/228728910-f16d09f0-3af3-4dec-9d25-5b6d253f7444.png">

Self invocation method return after the bar ends

pyongjie commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+   `printDeletedModule()` of the `Print` class. Otherwise, the `Parser` class calls upon `printNoDeletedModuleFound()`
+   of the `Print` class. This displays to the user the result of the `delete` command.
+
+![deleteModule](diagrams/DeleteModule.png)
```
Perhaps could shift the "[module not found]" in the alt frame more to the right as its overlapping with the activation bar?

sriram-senthilkr commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+
+The sequence of events above can be represented with the following sequence diagram:
+
+![calculateCAP](diagrams/CalculateCAP.png)
```
Good that self invocation bars are included. Should considering including them for the other diagrams too
<img width="245" alt="image" src="https://user-images.githubusercontent.com/88694646/228723677-000b61b4-9ab0-48ec-9f0f-acd9e37bd2c8.png">

pyongjie commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+
+The sequence of events above can be represented with the following sequence diagram:
+
+![gradeCommand](diagrams/GradeCommand.png)
```
Perhaps could have a better and clearer naming for the condition of the opt frame?

pyongjie commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
 
+### User Stories
 
-## Product scope
-### Target user profile
+| Version | As a...      | I want to...                                    | So that I can...                                                               |
+|---------|--------------|-------------------------------------------------|--------------------------------------------------------------------------------|
+| v1.0    | New user     | See usage instructions                          | Refer to them when I forget the commands                                       |
+| v1.0    | New user     | See usage instructions based on feature         | Get more in depth knowledge on the feature that I require                      |
+| v1.0    | New user     | Find a module by its code                       | Locate the module code without having to go through the entire list of modules |
+| v2.0    | Current user | Find a module by its type                       | Locate all modules in a category without having to go through the entire list  |
+| v1.0    | New user     | Add modules                                     | To plan out my 4 years in uni                                                  |
```
Perhaps can have a slightly more professional and better description for "So that I can"?

Sherlock-YH replied:
I like the layout

Sherlock-YH commented on docs/DeveloperGuide.md:
A good colour choice which makes the reader able to read with ease

Sherlock-YH commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+
+`ManPage#printManPageExit()` - Prints description of how to use the exit feature.
+
+![ManPage](diagrams/ManPage.png)
```
Maybe try to enlarge the diagram to improve readability

BernardLesley commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+
+{image}
+
+### 1. **Storage Component**
+
+Firstly, the {explain workflow}
+
+{uses}
+*  It can save the name of the user into a file.
+* The program can load data from the file when Modganiser starts up again.
+* It can save changes to the file after each change such as an addition or deletion.
+  This is done by using the methods
+
+
+### 2. **Parser Component**
+
+Firstly, the {explain workflow}
+
+{uses}
+* It calls the relevant methods according to the user input
+
+
+### 3. **ModuleList Component**
+
+Firstly, the {explain workflow}
+
+{uses}
+*
+
+### 4. **Print Component**
+
+
+Firstly, the {explain workflow}
+
+{uses}
+*
```
It seems that this section might be incomplete

BernardLesley commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+7) If the user wants to update the Semester, the `editSemester()` method is called.
+   This method directly modifies the `semester` attribute of the `Module` object.
+
+![editCommand](diagrams/EditCommand.png)
```
<img width="77" alt="image" src="https://user-images.githubusercontent.com/67412977/228729293-be8b35b8-9e64-4f78-b2e8-dbbdbd7148f7.png">
Add bars for self invocaton

BernardLesley reviewed (commented):
Overall great job on the developer guide, however the formatting as well as some of the sequence diagram can be improved.

SongZijin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+5. [Appendix: Instructions for manual testing](#appendix--instructions-for-manual-testing)
+
+
+## Introduction 
+
 
```
There is both introduction before table of contents, it bay be good to only have one

SongZijin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+{image}
+
+Main components of the architecture
+`Duke` has a class called `Duke`.
```
The name of the program is `Duke` here, which may be better if changed to `Modganiser`

SongZijin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+The `add` command is used by the user to add a module or multiple modules at once.
+
+The sequence by which the Parser class handles the `add` command is as follows:
+1) The `Parser` class extracts the necessary fields from the user input, and calls upon the `addModule`
+   method in the `ModuleList` class in a loop.
+2) The method then checks the type of module that is being added, then calls the relevant
+   constructor (`Core`, `UE`, `GE`, `Internship`).
+3) The `addModule` method returns the module that is added to the `Parser` class.
+4) The `getModuleListSize` method is called from `ModuleList` class.
+5) The `printAddedModule` method is called from the `Print` class to display the result to the user.
```
It may be better if the point form is used in a way such that it is properly seperated
```suggestion
1.  The `Parser` class extracts the necessary fields from the user input, and calls upon the `addModule` method in the `ModuleList` class in a loop.
2. The method then checks the type of module that is being added, then calls the relevant constructor (`Core`, `UE`, `GE`, `Internship`).
3. The `addModule` method returns the module that is added to the `Parser` class.
4. The `getModuleListSize` method is called from `ModuleList` class.
5. The `printAddedModule` method is called from the `Print` class to display the result to the user.
```

SongZijin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+feature
+
+The following object diagram shows the current state of the ChatBot:
+![FindModule](diagrams/FindTwoModules.png)
```
1. It may be clearer if the object that starts all of it is named `Modganiser` instead of `Duke`.
2. Since a Module object can contain attributes like: modular credits, year and semester, you may want to include them in the attribute section of the Module objects, since they are mentioned in step 2

SongZijin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,360 @@
+   and return the deleted `Module` object. Otherwise, the `deleteModule` method returns `null`.
+4) If a deleted `Module` object is returned from `deleteModule`, the `Parser` class calls upon
+   `printDeletedModule()` of the `Print` class. Otherwise, the `Parser` class calls upon `printNoDeletedModuleFound()`
+   of the `Print` class. This displays to the user the result of the `delete` command.
```
This section has the same formatting error as above, thus it may be better if it can be reformatted. This same mistake is also found in many other areas of the developer guide.

SongZijin reviewed (commented):
A very clear developer guide with multiple types of diagrams, however, there are some small issues with the formatting and the different name used for this programe.

chokyao committed:
Merge pull request #230 from anthea-pr0g/DG_diagrams

anthea- DG refined with sequence diagrams + architecture component

chokyao committed:
Merge branch 'master' into Update-DG

anthea-pr0g committed:
Merge pull request #233 from chokyao/Update-DG

Great job working on the DG!

saintzaw committed:
Add Parser Component to DG

saintzaw committed:
Added Documentation, Logging and Testing to DG

saintzaw committed:
Added more JUnit Tests

saintzaw committed:
Added Javadocs

saintzaw committed:
Reduced Line Length

chokyao committed:
Merge pull request #235 from saintzaw/Saint-UpdateDeveloperGuide

Saint updated Developer Guide, added Junit Tests and Javadocs

Thiolk committed:
Edit screenshot for edit module not in list warning to make it clearer

Thiolk committed:
Fix issue 227, make year 1.5 and 2.5 clearer

Thiolk committed:
Fix issue 209, changing hyperlink to contact us to the issues page
on github

saintzaw committed:
Merge pull request #236 from Thiolk/LengKiat-UserGuide

Update user guide to fix issues

Thiolk committed:
Fix issue 206

Change the print statement when user has completed
all the requirements for the module type

saintzaw committed:
Merge pull request #237 from Thiolk/LengKiat-TrackMC

Fix issue 206

saintzaw committed:
Saint Added More Checks for Edit Command

Thiolk committed:
Merge pull request #238 from saintzaw/Saint-FixEditBugs

Saint Added More Checks for Edit Command

Thiolk committed:
Fixes #208

Parser checks to ensure that user input is not empty

Thiolk committed:
Fixes #212

Extra error message will not be printed if print command is in try block
Error thrown will cause try block execution to stop

Thiolk committed:
Fixes #208

Change isEmpty to isBlank. isEmpty is not able to check if a string
contains only white spaces while isBlank can

geraldkoh4 committed:
Merge pull request #239 from Thiolk/LengKiat-CodeQuality

Fix bugs #208 and #212 found in PED

geraldkoh4 committed:
Fixed addCommand bugs, test cases and some print statements.

geraldkoh4 committed:
Minor bug fixes

saintzaw committed:
Merge pull request #241 from geraldkoh4/branch-Gerald-AddCommandFixes

Fixed addCommand bugs, test cases and some print statements.

Commit:
anthea- bugs to be fixed

Commit:
anthea- bugs to be fixed

geraldkoh4 committed:
Merge pull request #242 from anthea-pr0g/bugs

Bugs assigned from previous dry run

Commit:
anthea- PPP

Thiolk committed:
Merge pull request #243 from anthea-pr0g/diagrams

anthea- PPP

Commit:
anthea- PPP edits with screenshots

Thiolk committed:
Merge pull request #244 from anthea-pr0g/ppp

anthea- PPP edits with screenshots

Thiolk committed:
Update User Guide

Thiolk committed:
Fixes #225, #193

denzelcjy committed:
Edit add command to accept MCs from 0-41, and Year from 1-4

denzelcjy committed:
Fixed error message to be clearer

denzelcjy committed:
Edit editmc and edityear checks, changed list to module plan in print function

denzelcjy committed:
Merge pull request #246 from Thiolk/LengKiat-UserGuide

Update User Guide

denzelcjy committed:
Fix updated Junit Tests

Thiolk committed:
Merge pull request #248 from denzelcjy/DenzelChen-PEDCodeBugFixes

Denzel Chen PED code bug fixes

Thiolk committed:
Update Screenshots for user guife

Thiolk committed:
Fix issue 207

Thiolk committed:
Merge branch 'master' into LengKiat-UserGuide

denzelcjy committed:
Merge pull request #249 from Thiolk/LengKiat-UserGuide

Update Track command

Thiolk committed:
Update screenshot to latest for track command

geraldkoh4 committed:
Alphanumeric checks for add input

Thiolk committed:
Add new screenshot for track all command

denzelcjy committed:
Merge pull request #250 from Thiolk/LengKiat-UserGuide

Update to latest screenshot for track command

denzelcjy committed:
Merge pull request #251 from geraldkoh4/branch-Gerald-Alphanumeric

Alphanumeric checks for add input

chokyao committed:
Fix save file bug

chokyao committed:
Change function name in Print

denzelcjy committed:
Edited error message for wrong semester, updated JUnit test for it

chokyao committed:
Changed position of calls to saveModules

chokyao committed:
Refactor code

denzelcjy committed:
Fix `calculatecap` portion, other additional notes

chokyao committed:
Fix indentation bug

chokyao committed:
Fix bug

chokyao committed:
Fix bug

Commit:
anthea- PPP name

denzelcjy committed:
Update `track` command, update glossary

anthea-pr0g committed:
Merge pull request #253 from chokyao/Fix-Save-Bugs

Great work on fixing up the save file bug

Commit:
anthea- TA comments

chokyao committed:
Merge pull request #256 from denzelcjy/DenzelChen-FixUGBugs+MinorEdits

Denzel Chen Fix UG bugs + minor code edits

chokyao committed:
Merge pull request #257 from anthea-pr0g/TA

TA comments

Commit:
anthea- Track /all added in manpage

denzelcjy committed:
Merge pull request #261 from anthea-pr0g/track

anthea- Track /all added in manpage

Thiolk committed:
Refining the User Guide

Thiolk committed:
Edit User Guide to page break cleanly

Thiolk committed:
Improve page breaks in UG

Thiolk committed:
Update user guide

Thiolk committed:
Update PPP with recent contributions

Thiolk committed:
Update User Guide to ensure page breaks nicely

anthea-pr0g committed:
Merge pull request #262 from Thiolk/LengKiat-UserGuide

Nicely done!

Thiolk committed:
Refactor Praser checkUserInput method

denzelcjy committed:
First draft of PPP

chokyao committed:
Merge pull request #264 from denzelcjy/DenzelChen-PPP

Thiolk committed:
Fix task 1 of issue #259

Thiolk committed:
Fixes task 3 of issue 259

Refactor code to improve code quality

saintzaw committed:
Merge pull request #263 from Thiolk/LengKiat-CodeQuality

Refactor Parser checkUserInput method

Commit:
anthea- authorship

Thiolk committed:
Fix task 4 of issue 259

Refactor the findModuleByType method

Thiolk committed:
Merge pull request #265 from anthea-pr0g/author

anthea- authorship

anthea-pr0g committed:
Merge pull request #266 from Thiolk/LengKiat-CodeQuality

Great work fixing up the bugs.

Commit:
anthea- javadoc

Thiolk committed:
Merge branch 'master' into LengKiat-PPP

* master:
  Fix task 4 of issue 259
  anthea- authorship
  Fixes task 3 of issue 259
  Fix task 1 of issue #259
  First draft of PPP
  Refactor Praser checkUserInput method
  Update User Guide to ensure page breaks nicely
  Update user guide
  Improve page breaks in UG
  Edit User Guide to page break cleanly
  Refining the User Guide

Thiolk committed:
Added more contributions to PPP

Thiolk committed:
Merge pull request #267 from anthea-pr0g/javadoc

anthea- javadoc

Thiolk committed:
Add JavaDoc comments for new functions in Command class

saintzaw committed:
Saint Drafted PPP

saintzaw committed:
Merge pull request #269 from Thiolk/LengKiat-JavaDoc

Add JavaDoc comments for new functions in Command class

saintzaw committed:
Merge pull request #268 from Thiolk/LengKiat-PPP

Update PPP to include recent contributions

denzelcjy committed:
Merge pull request #270 from saintzaw/Saint-DraftPPP

Saint Drafted PPP

denzelcjy committed:
Make minor edits to UG

saintzaw committed:
Saint refactored confusing variables and added comments in printModuleListByYear method in Print.java to improve readability

saintzaw committed:
Saint refactored magic literals in InputChecker.java

Thiolk committed:
Merge pull request #271 from denzelcjy/FixUG

Denzel Chen Make minor edits to UG

saintzaw committed:
Saint refactored listModulesByYear method to a new method to reduce deep nesting

denzelcjy committed:
Update annotations

denzelcjy committed:
Update JavaDocs

saintzaw committed:
Saint Added Javadocs to Edit and List command methods in Command class

saintzaw committed:
Fixed indentation error

geraldkoh4 committed:
Merge pull request #272 from saintzaw/Saint-CodeReview

Saint fixed TA's code review issue and added Javadocs

denzelcjy committed:
Removed magic strings and abstracted into constants

denzelcjy committed:
More usage of constants

saintzaw committed:
Merge branch 'master' into DenzelChen-UpdateJavadocs+Annotations+CodeQuality

saintzaw committed:
Merge pull request #273 from denzelcjy/DenzelChen-UpdateJavadocs+Annotations+CodeQuality

Denzel Chen Update Javadocs+annotations+code quality

saintzaw committed:
Saint Added Authorship

chokyao committed:
Merge pull request #274 from saintzaw/Saint-Author

Saint Added Authorship

chokyao committed:
Add PPP draft

chokyao committed:
Update PPP

saintzaw committed:
Merge pull request #275 from chokyao/Add-PPP

Chok Yao PPP Draft

geraldkoh4 committed:
Fixed some issues that Vish brought up and authored. Updated PPP.

chokyao committed:
Merge branch 'master' into branch-Gerald-PPPAndAuthored

chokyao committed:
Merge pull request #276 from geraldkoh4/branch-Gerald-PPPAndAuthored

Fixed some issues that Vish brought up and authored. Updated PPP.

chokyao committed:
Edit Delete sequence diagram

anthea-pr0g committed:
Merge pull request #277 from chokyao/Edit-DG

Good job on cleaning up the DG

Commit:
anthea- DG updates based on TA

saintzaw committed:
Merge pull request #278 from anthea-pr0g/updated

anthea- DG updates based on TA

Commit:
anthea- DG updates based on TA (diagrams)

Commit:
anthea- DG updates based on TA (diagrams)

Thiolk committed:
Merge pull request #279 from anthea-pr0g/newdiag

anthea- DG updates based on TA (diagrams)

Thiolk committed:
Update PPP

saintzaw committed:
Merge pull request #280 from Thiolk/LengKiat-PPP

Update PPP

denzelcjy committed:
Updated puml for Grade Command

denzelcjy committed:
Update Grade Command diagram

denzelcjy committed:
update calculate cap command puml

denzelcjy committed:
update calculatecap command diagram

denzelcjy committed:
edit grade command diagram

denzelcjy committed:
Update DG for Grade and CalculateCAP commands

chokyao committed:
Merge pull request #281 from denzelcjy/DenzelChen-FixDGDiagrams

saintzaw committed:
Saint Updated the Parser Component in the DG

chokyao committed:
Merge pull request #282 from saintzaw/Saint-DG

chokyao committed:
Update sequence diagrams and other parts

chokyao committed:
Update PPP

saintzaw committed:
Merge pull request #283 from chokyao/Updates-to-DG

Chok Yao Update DG sequence diagrams and other parts

saintzaw committed:
Saint Updated DG

saintzaw committed:
Updated my PPP

denzelcjy committed:
Merge pull request #291 from saintzaw/Saint-FixDG

Saint Updated DG and PPP

denzelcjy committed:
extracted alphanumeric checker, implement for find, delete, grade commands

denzelcjy committed:
Added validateSetModuleGrade(), added javadocs, edited other functions and junit test that uses updateModuleGrade

denzelcjy committed:
Update grade command diagram

denzelcjy committed:
edit diagram for editModuleType

denzelcjy committed:
edit editModuleCode diagram

denzelcjy committed:
edit DG for grade command

denzelcjy committed:
edit User Guide for edit and grade commands

chokyao committed:
Merge pull request #292 from denzelcjy/DenzelChen-FixCommands+UG,DG-for-Grade

Commit:
anthea- manPage bugs

Commit:
anthea- manPage bugs (updated)

Commit:
anthea- manPage bugs (updated again)

geraldkoh4 committed:
Merge pull request #293 from anthea-pr0g/manpg

anthea- manPage bugs

Commit:
anthea- DG UI

denzelcjy committed:
Merge pull request #294 from anthea-pr0g/DG_UI

anthea- DG UI

Thiolk committed:
Add screenshots for new errors

Thiolk committed:
Edit screenshots and page breaks

Thiolk committed:
Edit AboutUs.md

Thiolk committed:
Edit PPP link for AboutUs

Thiolk committed:
Edit link to PPP file

Thiolk committed:
Edit all links to member's PPP pages

Thiolk committed:
Edit PPP

Thiolk committed:
Edit Saint PPP page link

chokyao committed:
Merge pull request #295 from Thiolk/LengKiat-UserGuide

Commit:
anthea- PPP final

Commit:
anthea- PPP final deletions

geraldkoh4 committed:
Merge pull request #297 from anthea-pr0g/new-ppp

New ppp

Commit:
anthea- PPP final look

geraldkoh4 committed:
Merge pull request #298 from anthea-pr0g/whitespace

anthea- PPP final look

Commit:
anthea- PPP final diagrams

anthea-pr0g committed:
Merge pull request #299 from anthea-pr0g/ppp-diagrams

anthea- PPP final diagrams

Thiolk committed:
Fix broken link

geraldkoh4 committed:
Merge pull request #300 from Thiolk/LengKiat-PPP

Fix broken link in PPP

Thiolk committed:
Replace image with notations

denzelcjy committed:
Merge pull request #301 from Thiolk/LengKiat-UserGuide

Replace image with one with notations

denzelcjy committed:
Edit PPP format and page break

denzelcjy committed:
Minor edits to PPP

denzelcjy committed:
add page break

Thiolk committed:
Change Cayman layout

denzelcjy committed:
changes to UG, PPP

Thiolk committed:
Place layout file in correct directory

Thiolk committed:
Add CSS style file

Thiolk committed:
Rename directory for css file

saintzaw committed:
Saint Added List Modules to Design and Implementation in DG

denzelcjy committed:
edit page breaks for UG

saintzaw committed:
Edited PPP

Thiolk committed:
Edit header and header css

Thiolk committed:
Adjust header padding

geraldkoh4 committed:
Merge pull request #302 from saintzaw/Saint-ListCommandDG

Saint Added List Command to DG

geraldkoh4 committed:
Merge pull request #303 from denzelcjy/DenzelChen-UpdatePPP

Denzel Chen Update PPP and UG

Thiolk committed:
Add line break after header

Thiolk committed:
Add page breaks for all pages after banner

geraldkoh4 committed:
Merge pull request #304 from Thiolk/LengKiat-UserGuide

Update theme of all documents

geraldkoh4 committed:
Updated PPP, DG, and add command Error messages

geraldkoh4 committed:
Fixed junit testing

saintzaw committed:
Merge pull request #305 from geraldkoh4/branch-Gerald-PPP,DG,ErrorMessage,Updates

Updated PPP, DG, and add command Error messages

saintzaw committed:
Saint moved check for blank user input, added logging for edit

saintzaw committed:
Saint added sequence diagram for list command

denzelcjy committed:
Finalise PPP

saintzaw committed:
Fixed Junit test, made changes to DG

saintzaw committed:
Saint made minor changes (fixed typos) to UG

chokyao committed:
Merge pull request #306 from saintzaw/Saint-DG,Updates

Saint updated DG and blank user input checks

saintzaw committed:
Merge pull request #307 from denzelcjy/DenzelChen-FinalisePPP

Denzel Chen Finalise PPP

Thiolk committed:
Update user guide

Thiolk committed:
Updated UG

Thiolk committed:
Change error message to match UG

chokyao committed:
Merge pull request #308 from Thiolk/LengKiat-UserGuide

Thiolk committed:
Add line to maintain consistency in UG

denzelcjy committed:
Merge pull request #309 from Thiolk/LengKiat-UserGuide

Update UG

saintzaw committed:
Saint updated DG and PPP

chokyao committed:
Merge pull request #310 from saintzaw/Saint-PPP,DG

Thiolk committed:
Edit title

denzelcjy committed:
Merge pull request #311 from Thiolk/LengKiat-UserGuide

Edit title of documents

Thiolk committed:
Add additional information to find feature

Thiolk committed:
Add line break

denzelcjy committed:
Merge pull request #312 from Thiolk/LengKiat-DG

Add additional information to find feature in DG

saintzaw committed:
Saint added track command to DG

chokyao committed:
Merge pull request #313 from saintzaw/Saint-PPP,DG,Updates

Saint added track command to DG

chokyao committed:
Add trackCommand sequence diagram

chokyao committed:
Update formatting of DG

chokyao committed:
Update PPP

chokyao committed:
Changes to PPP

saintzaw committed:
Merge pull request #314 from chokyao/Changes-to-DG

Updated DG formatting and seq diagram for track

chokyao committed:
Add page breaks

anthea-pr0g committed:
Merge pull request #315 from chokyao/Edits-to-DG

good job on formatting the DG :)

Thiolk committed:
Made edits to page breaks and track command

chokyao committed:
Merge pull request #316 from Thiolk/LengKiat-DG

Made edits to page breaks and track command

chokyao committed:
Update PPP format and DG page break

saintzaw committed:
Merge pull request #317 from chokyao/Edit-to-PPP

Chok Yao Update PPP

chokyao committed:
Update PPP

saintzaw committed:
Merge pull request #318 from chokyao/Update-to-PPP

Update PPP

saintzaw committed:
Saint updated PPP and fixed a typo in DG track command

geraldkoh4 committed:
FinalisedPPP

chokyao committed:
Merge pull request #319 from saintzaw/Saint-FinalisePPP

saintzaw committed:
Merge pull request #320 from geraldkoh4/branch-Gerald-PPPFinalised

FinalisedPPP

saintzaw committed:
Saint Updated PPP

saintzaw committed:
Updated PPP

geraldkoh4 committed:
Updated PPP due to github pages formatting issues

chokyao committed:
Merge pull request #321 from saintzaw/Saint-UpdatePPP

saintzaw committed:
Edited page breaks

saintzaw committed:
Merge pull request #323 from saintzaw/Saint-FinalPPP

Edited page breaks

saintzaw committed:
Merge pull request #322 from geraldkoh4/branch-Gerald-PPPUpdated

Updated PPP due to github pages formatting issues

chokyao committed:
Update storage diagram

geraldkoh4 committed:
Merge pull request #325 from chokyao/Update-DG-diagram

Chok Yao Update storage diagram

saintzaw committed:
Saint Updated Parser Class Diagram in DG

saintzaw committed:
Updated DG

chokyao committed:
Merge pull request #326 from saintzaw/Saint-EditParserDiagram

Saint edited parser class diagram in DG

geraldkoh4 committed:
Updated DG diagram

geraldkoh4 committed:
Changed the diagram to the updated version

saintzaw committed:
Merge pull request #327 from geraldkoh4/branch-Gerald-DGdiagramUpdate

Branch gerald dg diagram update

geraldkoh4 committed:
Updated PPP for submission

saintzaw committed:
Merge pull request #328 from geraldkoh4/branch-Gerald-UpdatePPP

Updated PPP for submission

geraldkoh4 committed:
Updated page break for PPP

Thiolk committed:
Merge pull request #329 from geraldkoh4/branch-Gerald-FixedPPP

Updated page break for PPP
